pkg/connector/matrix.go

```diff
@@ -293,23 +299,52 @@ func (t *TelegramClient) PreHandleMatrixReaction(ctx context.Context, msg *bridg
 		return resp, err
 	}
 
-	var emojiID networkid.EmojiID
+	keyNoVariation := variationselector.Remove(msg.Content.RelatesTo.Key)
+	emojiID := ids.MakeEmojiIDFromEmoticon(msg.Content.RelatesTo.Key)
 	if strings.HasPrefix(msg.Content.RelatesTo.Key, "mxc://") {
 		if file, err := t.main.Store.TelegramFile.GetByMXC(ctx, msg.Content.RelatesTo.Key); err != nil {
 			return resp, err
 		} else if file == nil {
+			// Remove the reaction if it's not allowed
+			t.main.Bridge.QueueRemoteEvent(t.userLogin, &simplevent.Reaction{
+				EventMeta: simplevent.EventMeta{
+					Type: bridgev2.RemoteEventReactionRemove,
+					LogContext: func(c zerolog.Context) zerolog.Context {
+						return c.
+							Str("key", msg.Content.RelatesTo.Key).
+							Str("target", string(msg.TargetMessage.ID))
+					},
+					PortalKey: msg.Portal.PortalKey,
+				},
+				TargetMessage: msg.TargetMessage.ID,
+				EmojiID:       ids.MakeEmojiIDFromEmoticon(msg.Content.RelatesTo.Key),
+				Emoji:         msg.Content.RelatesTo.Key,
+			})
```

Comment:
This should definitely not be here, removing reactions can be done in mautrix-go